credentials/alts/alts_test.go

```diff
@@ -287,6 +296,50 @@ func (s) TestCheckRPCVersions(t *testing.T) {
 	}
 }
 
+func TestFullHandshake(t *testing.T) {
+	// If GOMAXPROCS is set to less than 2, do not run this test. This test
+	// requires at least 2 goroutines to succeed (one goroutine where a
+	// server listens, another goroutine where a client runs).
+	if runtime.GOMAXPROCS(0) < 2 {
+		return
+	}
+
+	// Start the fake handshaker service and the server.
+	var wait sync.WaitGroup
+	defer wait.Wait()
+	stopHandshaker, handshakerAddress := startFakeHandshakerService(t, &wait)
+	defer stopHandshaker()
+	stopServer, serverAddress := startServer(t, handshakerAddress, &wait)
+	defer stopServer()
+
+	// Connect to the server, authenticating with ALTS and the fake handshaker service.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	vmOnGCP = true
+	creds := NewClientCreds(&ClientOptions{HandshakerServiceAddress: handshakerAddress})
+	dialOptions := []grpc.DialOption{
+		grpc.WithTransportCredentials(creds),
+		grpc.WithBlock(),
+		grpc.WithReturnConnectionError(),
+	}
+	conn, err := grpc.DialContext(ctx, serverAddress, dialOptions...)
+	if err != nil {
+		t.Fatalf("grpc.Dial(%v, %v) failed: %v", serverAddress, dialOptions, err)
+	}
+	defer conn.Close()
+
+	// Ping the server.
+	c := testpb.NewTestServiceClient(conn)
+	if _, err = c.UnaryCall(context.Background(), &testpb.SimpleRequest{}); err != nil {
```

Comment:
Prefer using `grpc.Dial` above and use a context with a reasonable deadline here.

Reply:
Done.